buildimage.go

```diff
@@ -36,6 +36,14 @@ var (
 	wifi    = flag.Bool("wifi", true, "include wifi tools")
 )
 
+//precheck can check anything from go env
```

Comment:
this is not checking anything. It's just a wrapper for os.Getenv

Why not just warn users but let them do what they want?

// preCheck checks the value of an environment variable against a set of allowed values and returns
// an error if none of them match.
func preCheck(val string, allowed ...string) error {
// you may or may not have seen this style of declaration. 
}

Reply:
why “suggest”. there’s no point of suggesting to export it as off if it won’t work for them with it being off. or on. i think it should be set as auto or “”, or program exits

Reply:
Yes. Check it for a proper value and exit with a warning. if it is wrong.  I am pretty sure that's what I said.

Or see how the u-root command sets up CGO_ENABLED. You need not check; just set the proper value of GO111MODULE in the Env array of the exec.Cmd structs you use to run programs. If you are not familiar with how the Env array works in the Cmd struct it's time to take a look.